libvirt/data_source_libvirt_network.go

```diff
@@ -156,3 +156,76 @@ func resourceLibvirtNetworkDNSSRVRead(d *schema.ResourceData, meta interface{})
 
 	return nil
 }
+
+//// DHCP
+
+// a libvirt network DHCP host template datasource
+//
+// Datasource example:
+//
+// data "libvirt_network_dhcp_host_template" "k8smasters" {
+//   count = "${var.master_count}"
+//   ip = "${var.master_ips[count.index]}"
+//   mac = "${var.master_mac[count.index]}"
+//   name = "master-${count.index}"
+// }
+//
+// resource "libvirt_network" "k8snet" {
+//   ...
+//   dhcp = [{
+//     hosts = [ "${flatten(data.libvirt_network_dhcp_host_template.k8smasters.*.rendered)}" ]
+//   }]
+//   ...
+// }
+//
+func datasourceLibvirtNetworkDHCPHostTemplate() *schema.Resource {
+	return &schema.Resource{
+		Read: resourceLibvirtNetworkDHCPHostRead,
+		Schema: map[string]*schema.Schema{
+			"ip": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"mac": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
```

Comment:
Is this a hostname? If that is the case, maybe we should use `hostname` (as we do in the `libvirt_network_dns_host_template`)

Reply:
Indeed a hostname, I used `name` as used in libvirt - https://github.com/libvirt/libvirt-go-xml/blob/master/network.go#L145